core/src/main/java/io/aiven/klaw/validation/TopicRequestValidatorImpl.java

```diff
@@ -1,21 +1,244 @@
+import io.aiven.klaw.service.TopicControllerService;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 
 @Slf4j
 public class TopicRequestValidatorImpl
     implements ConstraintValidator<TopicRequestValidator, TopicRequestModel> {
 
+  @Autowired private CommonUtilsService commonUtilsService;
+  @Autowired private MailUtils mailService;
+  @Autowired private TopicControllerService topicControllerService;
+
+  private PermissionType permissionType;
+
   @Override
-  public void initialize(TopicRequestValidator constraintAnnotation) {}
+  public void initialize(TopicRequestValidator constraintAnnotation) {
+    this.permissionType = constraintAnnotation.getPermissionType();
+  }
 
   @Override
   public boolean isValid(
       TopicRequestModel topicRequestModel, ConstraintValidatorContext constraintValidatorContext) {
+
+    // Verify if user has access to request for topics
+    if (commonUtilsService.isNotAuthorizedUser(
+        topicControllerService.getPrincipal(), this.permissionType)) {
+      updateConstraint(constraintValidatorContext, ApiResultStatus.NOT_AUTHORIZED.value);
+      return false;
+    }
+
+    // tenant filtering
+    if (!topicControllerService
+        .getEnvsFromUserId(topicControllerService.getUserName())
+        .contains(topicRequestModel.getEnvironment())) {
+      updateConstraint(
+          constraintValidatorContext,
+          "Failure. Not authorized to request topic for this environment.");
+      return false;
+    }
+
+    // check for null topic name
+    if (topicRequestModel.getTopicname() == null) {
+      updateConstraint(constraintValidatorContext, "Failure. Please fill in topic name.");
+      return false;
+    }
+
+    // check for empty topic name
+    if (topicRequestModel.getTopicname().length() == 0) {
+      updateConstraint(constraintValidatorContext, "Failure. Please fill in a valid topic name.");
+      return false;
+    }
+
+    // verify tenant config exists
+    int tenantId = commonUtilsService.getTenantId(topicControllerService.getUserName());
+    String syncCluster;
+    try {
+      syncCluster = topicControllerService.getSyncCluster(tenantId);
+    } catch (Exception e) {
+      log.error("Tenant Configuration not found. " + tenantId, e);
+      updateConstraint(
+          constraintValidatorContext,
+          "Failure. Tenant configuration in Server config is missing. Please configure.");
+      return false;
+    }
+
+    // Check if topic is owned by a different team, applicable on update requests
+    List<Topic> topics =
+        topicControllerService.getTopicFromName(topicRequestModel.getTopicname(), tenantId);
+    if (topics != null
+        && topics.size() > 0
+        && !Objects.equals(
+            topics.get(0).getTeamId(),
```

Comment:
Didn't get it..
Why do we check only the first element?
What if order is changing somewhere?
Or does it have same value for all elements?
I guess a clarification comment in the code is required

Reply:
Added comment. As there could be only one owner team for topic, with topic name being unique in a tenant, any element check is enough.